AK/Optional.h

```diff
@@ -185,6 +185,119 @@ class [[nodiscard]] Optional {
     alignas(T) u8 m_storage[sizeof(T)];
     bool m_has_value { false };
 };
+
+template<typename T>
+class [[nodiscard]] Optional<T&> {
+public:
+    using ValueType = T&;
+
+    ALWAYS_INLINE Optional() = default;
+
+    ALWAYS_INLINE Optional(Optional const& other) = delete;
```

Comment:
You didn't need to delete this one. Reference creation is shallow as well, so there can be no inconsistency here. The inconsistensies can only arise when we try to change the state of the Optional after it has been created.